src/JsonSchema/Uri/UriRetriever.php

```diff
@@ -25,6 +25,14 @@
 class UriRetriever implements BaseUriRetrieverInterface
 {
     /**
+     * @var array Map of URL translations
+     */
+    protected $translationMap = array(
+        // use local copies of the spec schemas
+        '|^https?://json-schema.org/draft-(0[34])/schema#?|' => 'package://dist/schema/json-schema-draft-$1.json'
```

Comment:
I'm going to bow out of this one. Pick your battles, and all that. Conversationally, I will say that it's not uncommon in the slightest for external libraries to do local writes. I have 3 or 4 in my own project now that do this (php minify, sass, and typescript--to name a few off the top of my head--all have cache directories). And I can think of other possible scenarios where it would be useful even in this one; one idea I've been mulling lately is the concept of machine-generating--and caching--a validator function for any given schema that is ugly as hell but super speedy (though I kind of suspect that would entail starting over from scratch). 

LGTM